operators/multiclusterobservability/controllers/multiclusterobservability/observatorium.go

```diff
@@ -159,20 +159,22 @@ func GenerateObservatoriumCR(
 
 	oldSpec := observatoriumCRFound.Spec
 	newSpec := observatoriumCR.Spec
-	oldSpecBytes, _ := yaml.Marshal(oldSpec)
-	newSpecBytes, _ := yaml.Marshal(newSpec)
-	if bytes.Equal(newSpecBytes, oldSpecBytes) &&
-		labels[obsCRConfigHashLabelName] == observatoriumCRFound.Labels[obsCRConfigHashLabelName] {
-		return nil, nil
-	}
 
-	// keep the tenant id unchanged
+	// keep the tenant id unchanged and ensure the new spec has the same tenant ID as the old spec to prevent Observatorium
+	// from updating
 	for i, newTenant := range newSpec.API.Tenants {
 		for _, oldTenant := range oldSpec.API.Tenants {
 			updateTenantID(&newSpec, newTenant, oldTenant, i)
 		}
 	}
 
+	oldSpecBytes, _ := yaml.Marshal(oldSpec)
+	newSpecBytes, _ := yaml.Marshal(newSpec)
+	if bytes.Equal(newSpecBytes, oldSpecBytes) &&
+		labels[obsCRConfigHashLabelName] == observatoriumCRFound.Labels[obsCRConfigHashLabelName] {
+		return nil, nil
```

Comment:
Through no fault of your own, but we're inconsistent with error handling in this reconcile loop. We alternate between returning an empty string `""` and the `nil` type. Later we pass the actual value to the `obsCRConfigHashLabelName`, do we know how this `nil` is handled in this case?